src/main/kotlin/org/rust/lang/core/psi/ext/RsArrayType.kt

```diff
@@ -18,9 +17,42 @@ val RsArrayType.arraySize: Int? get() {
     return calculateArraySize(expr)
 }
 
-// TODO: support constants and compile time expressions
-fun calculateArraySize(expr: RsExpr?): Int? = (expr as? RsLitExpr)
-    ?.integerLiteral
-    ?.text
-    ?.removeSuffix(TyInteger.Kind.usize.name)
-    ?.toIntOrNull()
+fun calculateArraySize(expr: RsExpr?): Int? = tryEvaluateExpr(expr)
+
+private fun tryEvaluateExpr(expr: RsExpr?): Int? {
+    return when (expr) {
+        is RsLitExpr -> expr.integerLiteral
+            ?.text
+            ?.removeSuffix(TyInteger.Kind.usize.name)
+            ?.toIntOrNull()
+            ?.notNegativeOrNull()
+        is RsPathExpr -> {
+            val const = expr.path.reference.resolve() as? RsConstant ?: return null
+            val integerType = const.typeReference?.type as? TyInteger ?: return null
```

Comment:
Maybe use `TyPrimitive.fromPath` directly instead of full type inference?

Reply:
Umm... maybe =D